etc/systemd/system/storm-webdav.service

```diff
@@ -6,7 +6,7 @@ WorkingDirectory=/etc/storm/webdav
 EnvironmentFile=-/etc/sysconfig/storm-webdav
 User=storm
 Type=simple
-ExecStart=/bin/bash -ac "/usr/bin/java $STORM_WEBDAV_JVM_OPTS -Djava.io.tmpdir=/var/lib/storm-webdav/work -Dlogging.config=$STORM_WEBDAV_LOG_CONFIGURATION -jar $STORM_WEBDAV_JAR"
+ExecStart=/bin/bash -ac "exec /usr/bin/java $STORM_WEBDAV_JVM_OPTS -Djava.io.tmpdir=/var/lib/storm-webdav/work -Dlogging.config=$STORM_WEBDAV_LOG_CONFIGURATION -jar $STORM_WEBDAV_JAR > ${STORM_WEBDAV_OUT} 2>${STORM_WEBDAV_ERR}"
```

Comment:
we cannot use directly the combination of:

```
ExecStart=/usr/bin/java $STORM_WEBDAV_JVM_OPTS -Djava.io.tmpdir=/var/lib/storm-webdav/work -Dlogging.config=$STORM_WEBDAV_LOG_CONFIGURATION -jar $STORM_WEBDAV_JAR
StandardOutput=${STORM_WEBDAV_OUT}
StandardError=${STORM_WEBDAV_ERR}
```

because the environment variables wouldn't be resolved in the ExecStart command.
The `/bin/bash -ac "<command>"` solution also allows us to pipe something at the end. So, we can redirect to file stdout and stderr. We use `exec` to replace shell with the target process. By using this, we will get rid of need for killing 2 processes.

Without exec, in fact, the status of the process would be:

```
# systemctl status storm-webdav
storm-webdav.service - StoRM WebDAV service
   Loaded: loaded (/usr/lib/systemd/system/storm-webdav.service; enabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/storm-webdav.service.d
           └─filelimit.conf
   Active: active (running) since Mon 2020-05-18 07:55:56 UTC; 9s ago
 Main PID: 8935 (bash)
   CGroup: /docker/40ba99ef607a936f01d0073d614fd2a5ed52a92f4a5818a0d8e1f99f490fd036/system.slice/storm-webdav.service
           ├─8935 /bin/bash -ac /usr/bin/java $STORM_WEBDAV_JVM_OPTS -Djava.io.tmpdir=/var/lib/storm-webdav/work -Dlogging.config=$STORM_WEBDAV_LOG_CONFIGURATION -jar $STORM_WEBDA...
           └─8936 /usr/bin/java -Xms256m -Xmx512m -Djava.io.tmpdir=/var/lib/storm-webdav/work -Dlogging.config=/etc/storm/webdav/logback.xml -jar /usr/share/java/storm-webdav/stor...
```

Two processes. The main is a bash process, the second is a java process.
With the added exec, instead:

```
# systemctl status storm-webdav
storm-webdav.service - StoRM WebDAV service
   Loaded: loaded (/usr/lib/systemd/system/storm-webdav.service; enabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/storm-webdav.service.d
           └─filelimit.conf
   Active: active (running) since Mon 2020-05-18 07:59:18 UTC; 1s ago
 Main PID: 9000 (java)
   CGroup: /docker/40ba99ef607a936f01d0073d614fd2a5ed52a92f4a5818a0d8e1f99f490fd036/system.slice/storm-webdav.service
           └─9000 /usr/bin/java -Xms256m -Xmx512m -Djava.io.tmpdir=/var/lib/storm-webdav/work -Dlogging.config=/etc/storm/webdav/logback.xml -jar /usr/share/java/storm-webdav/stor...

May 18 07:59:18 webdav.test.example systemd[1]: Started StoRM WebDAV service.
```

We can see only the java process.

Reply:
Thanks for the clarification